test/E2ETest/Microsoft.Test.E2E.AspNet.OData/QueryComposition/SelectExpandEFTests.cs

```diff
@@ -0,0 +1,349 @@
+
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, queryUrl);
+            request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json"));
+            HttpClient client = new HttpClient() { Timeout = TimeSpan.FromMinutes(10) };
+            HttpResponseMessage response;
+
+            // Act
+            response = await client.SendAsync(request);
+
+            // Assert
+            Assert.NotNull(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.NotNull(response.Content);
+
+            JObject responseObject = JObject.Parse(await response.Content.ReadAsStringAsync());
+            JArray result = responseObject["value"] as JArray;
+            Assert.Single(result);
+            Assert.Equal("Prop0001", result[0]["Custom1"]["Prop0001"]);
+            Assert.Equal("Prop0099", result[0]["Custom1"]["Prop0099"]);
+            Assert.Equal("Prop0199", result[0]["Custom1"]["Prop0199"]);
+            Assert.Equal("Prop0298", result[0]["Custom1"]["Prop0298"]);
+            Assert.Equal("Prop0798", result[0]["Custom2"]["Prop0798"]);
+            Assert.Equal("Prop1198", result[0]["Custom3"]["Prop1198"]);
+            Assert.Equal("Prop1598", result[0]["Custom4"]["Prop1598"]);
+            Assert.Equal("Prop1998", result[0]["Custom5"]["Prop1998"]);
+            Assert.Null(result[0]["Custom5"]["Prop2000"]);
+        }
+
+        private async Task RestoreData(string suffix)
+        {
+            string requestUri = BaseAddress + string.Format("/selectexpand/ResetDataSource{0}", suffix);
+            HttpClient client = new HttpClient();
+            HttpResponseMessage response = await client.GetAsync(requestUri);
+            response.EnsureSuccessStatusCode();
+        }
+    }
+
+    public class EFSelectCustomersController : TestODataController
+    {
+        private readonly SampleContext _db = new SampleContext();
+
+        [EnableQuery(PageSize = 2)]
+        public ITestActionResult Get()
+        {
+            return Ok(_db.Customers);
+        }
+
+        [HttpGet]
+        [ODataRoute("ResetDataSource-Customer")]
+        public ITestActionResult ResetDataSource()
+        {
+            _db.Database.Delete();  // Start from scratch so that tests aren't broken by schema changes.
+            Generate(_db);
+            return Ok();
+        }
+
+        public static void Generate(SampleContext db)
+        {
+            var customer = new EFSelectCustomer
+            {
+                Id = 1,
+                SelectOrders = new List<EFSelectOrder>
+                {
+                    new EFSelectOrder
+                    {
+                        Id = 3,
+                    },
+                    new EFSelectOrder
+                    {
+                        Id = 1,
+                    },
+                    new EFSelectOrder
+                    {
+                        Id = 2,
+                    }
```

Comment:
Do you need to set these order instance into db.Orders?

Reply:
This method was just moved from another file. I didn't change it. Presumably it works, because the tests pass.